test/cql-pytest/test_tools.py

```diff
@@ -770,6 +770,192 @@ def test_fail_nonexistent_table(self, scylla_path, system_scylla_local_sstable_p
                 error_msg="error processing arguments: could not load schema via schema-tables: std::runtime_error (Failed to find system.non-existent-table in schema tables)")
 
 
+@pytest.fixture(scope="class")
+def schema_test_base_table(cql, test_keyspace):
+    with util.new_test_table(cql, test_keyspace, "pk int, v1 text, v2 text, PRIMARY KEY (pk)") as table:
+        yield table
+
+
+@pytest.fixture(scope="class")
+def schema_test_mv(cql, schema_test_base_table):
+    with util.new_materialized_view(cql, schema_test_base_table, '*', 'v1, pk', 'v1 is not null and pk is not null') as mv:
+        yield mv
+
+
+@pytest.fixture(scope="class")
+def schema_test_si(cql, schema_test_base_table):
+    keyspace, base_table = schema_test_base_table.split(".")
+    si_name = f"{base_table}_by_v2"
+    cql.execute(f"CREATE INDEX {si_name} ON {schema_test_base_table}(v2)")
+    try:
+        yield f"{keyspace}.{si_name}_index"
+    finally:
+        cql.execute(f"DROP INDEX {keyspace}.{si_name}")
```

Comment:
By the way, just like you used above util.new_test_table() and util.new_materialized_view(), there is also a util.new_secondary_index(). It yields keyspace.name, so you just add the "_index" to what it returns. Any reason not to use it?

Reply:
I don't remember any particular reason, maybe I just missed it.

Reply:
Using `new_secondary_index()` in v2.